server/src/main/java/org/opensearch/action/update/TransportUpdateAction.java

```diff
@@ -32,121 +32,35 @@
-import org.opensearch.action.support.single.instance.TransportInstanceSingleOperationAction;
-import org.opensearch.cluster.ClusterState;
-import org.opensearch.cluster.metadata.IndexNameExpressionResolver;
+import org.opensearch.action.support.HandledTransportAction;
+import org.opensearch.action.support.WriteRequest;
 import org.opensearch.cluster.metadata.Metadata;
-import org.opensearch.cluster.routing.PlainShardIterator;
-import org.opensearch.cluster.routing.ShardIterator;
-import org.opensearch.cluster.routing.ShardRouting;
-import org.opensearch.cluster.service.ClusterService;
-import org.opensearch.common.collect.Tuple;
 import org.opensearch.common.inject.Inject;
-import org.opensearch.common.logging.DeprecationLogger;
-import org.opensearch.common.settings.Settings;
-import org.opensearch.common.xcontent.XContentHelper;
 import org.opensearch.core.action.ActionListener;
-import org.opensearch.core.common.bytes.BytesReference;
-import org.opensearch.core.common.io.stream.NotSerializableExceptionWrapper;
-import org.opensearch.core.common.io.stream.StreamInput;
-import org.opensearch.core.index.shard.ShardId;
-import org.opensearch.core.rest.RestStatus;
-import org.opensearch.core.xcontent.MediaType;
-import org.opensearch.index.IndexNotFoundException;
-import org.opensearch.index.IndexService;
-import org.opensearch.index.IndexSettings;
-import org.opensearch.index.engine.VersionConflictEngineException;
-import org.opensearch.index.shard.IndexShard;
-import org.opensearch.index.shard.IndexingStats.Stats.DocStatusStats;
-import org.opensearch.indices.IndicesService;
 import org.opensearch.tasks.Task;
-import org.opensearch.threadpool.ThreadPool;
-import org.opensearch.threadpool.ThreadPool.Names;
 import org.opensearch.transport.TransportService;
-import org.opensearch.transport.client.node.NodeClient;
-
-import java.io.IOException;
-import java.util.Collections;
-import java.util.Map;
-
-import static org.opensearch.ExceptionsHelper.unwrapCause;
-import static org.opensearch.action.bulk.TransportSingleItemBulkWriteAction.toSingleItemBulkRequest;
-import static org.opensearch.action.bulk.TransportSingleItemBulkWriteAction.wrapBulkResponse;
 
 /**
- * Transport action for updating an index
+ * Performs the update operation.
+ * <p>
+ * Deprecated use TransportBulkAction with a single item instead
  *
  * @opensearch.internal
  */
-public class TransportUpdateAction extends TransportInstanceSingleOperationAction<UpdateRequest, UpdateResponse> {
-    private static final DeprecationLogger deprecationLogger = DeprecationLogger.getLogger(TransportUpdateAction.class);
-    private final AutoCreateIndex autoCreateIndex;
-    private final UpdateHelper updateHelper;
-    private final IndicesService indicesService;
-    private final NodeClient client;
-    private final ClusterService clusterService;
+@Deprecated
+public class TransportUpdateAction extends HandledTransportAction<UpdateRequest, UpdateResponse> {
```

Comment:
Could we remove this class all together? Seems to not be widely used: https://github.com/search?q=org%3Aopensearch-project%20TransportUpdateAction%20NOT%20is%3Aarchived&type=code

Reply:
So, the `@Deprecated` annotation was copied from `TransportIndexAction` and `TransportDeleteAction`.

In practice, I don't think I actually want to remove it. I should probably remove the annotation. (We should probably remove the annotations from `TransportIndexAction` and `TransportDeleteAction` since I don't think they're ever going to be removed either.)

Reply:
There -- I removed this annotation and the ones on `TransportIndexAction`, `TransportDeleteAction`, and `TransportSingleItemBulkAction`.